### PR TITLE
Add support for severity options in AlertPolicy

### PR DESCRIPTION
**Pull request by jedouard1994**
Resolves https://github.com/hashicorp/terraform-provider-google/issues/16627

Adds support for severity options in AlertPolicy.

I acknowledge that I have:

* Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
* Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
* [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [make test and make lint](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
* [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
* Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
monitoring: added `severity` field to `google_monitoring_alert_policy` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go [hunk 1]
```diff
@@ -319,6 +319,8 @@ resource "google_monitoring_alert_policy" "full" {
     }
   }
 
+  severity     = "WARNING"
```
Comment (roaks3) [inline mmv1/third_party/terraform/services/monitoring/resource_monitoring_alert_policy_test.go:322]: Would it be possible to also include this in the basicCfg test, and demonstrate that the field can be changed eg. `ERROR` -> `WARNING`?

**Review by roaks3 — COMMENTED** [review]
LGTM, but the field appears to be updatable based on the config, so we should probably test that if possible.

**Comment by jedouard1994** [comment]
@roaks3 Sorry for the delay. Update tests were failing in prod until a recent deployment. 

They've been added and should be passing now.

**Comment by roaks3** [comment]
It looks like the linter is unhappy with the formatting of one of the files. Could you double check?
```
Run make lint
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
./google-beta/services/monitoring/resource_monitoring_alert_policy_test.go
You can use the command: `make fmt` to reformat code.
```

Also, I missed this before, but can we make sure what the behavior is if the `severity` is not specified? If there is a default set by the service, we may need to add `default_from_api: true`.

**Comment by jedouard1994** [comment]
Reformatted the go files, hopefully this fixes linter.

If severity is not specified, the API omits the field. There is no default.

**Review by roaks3 — APPROVED** [review]
LGTM

**Comment by roaks3** [comment]
Note: I tweaked the issue description to include the full issue URL that this PR fixes. That will trigger the auto-close behavior.

**Pull request merged**